fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java

```diff
@@ -1478,6 +1498,185 @@ public void addProperty(String pair) {
         }
     }
 
+    /**
+     * Perform the special data migration steps required for the V0010 version of the schema
+     */
+    protected void applyDataMigrationForV0010() {
+        if (MULTITENANT_FEATURE_ENABLED.contains(dbType)) {
+            // Process each tenant one-by-one
+            List<TenantInfo> tenants = getTenantList();
+            for (TenantInfo ti: tenants) {
+
+                // If no --schema-name override was specified, we process all tenants, otherwise we
+                // process only tenants which belong to the override schema name
+                if (!schema.isOverrideDataSchema() || schema.matchesDataSchema(ti.getTenantSchema())) {
+                    dataMigrationForV0010(ti);
+                }
+            }
+        } else {
+            doMigrationForV0010();
+        }
+
+    }
+
+    /**
+     * Migrate the IS_DELETED data for the given tenant
+     * @param ti
+     */
+    private void dataMigrationForV0010(TenantInfo ti) {
+        // Multi-tenant schema so we know this is Db2:
+        Db2Adapter adapter = new Db2Adapter(connectionPool);
+
+        Set<String> resourceTypes = Arrays.stream(FHIRResourceType.ValueSet.values())
+                .map(FHIRResourceType.ValueSet::value)
+                .collect(Collectors.toSet());
+
+        // Process each update in its own transaction so we don't stress the tx log space
+        for (String resourceTypeName: resourceTypes) {
+            try (ITransaction tx = TransactionFactory.openTransaction(connectionPool)) {
+                try {
+                    SetTenantIdDb2 setTenantId = new SetTenantIdDb2(schema.getAdminSchemaName(), ti.getTenantId());
+                    adapter.runStatement(setTenantId);
+
+                    GetLogicalResourceIsDeletedNeedsMigration needsMigrating = new GetLogicalResourceIsDeletedNeedsMigration(schema.getSchemaName(), resourceTypeName);
+                    if (adapter.runStatement(needsMigrating)) {
+                        logger.info("V0010 Migration: Updating " + resourceTypeName + "_LOGICAL_RESOURCES.IS_DELETED "
+                                + "for tenant '" + ti.getTenantName() + "', schema '" + ti.getTenantSchema() + "'");
+                        InitializeLogicalIdIsDeleted cmd = new InitializeLogicalIdIsDeleted(schema.getSchemaName(), resourceTypeName);
+                        adapter.runStatement(cmd);
+                    }
+                } catch (DataAccessException x) {
+                    // Something went wrong, so mark the transaction as failed
+                    tx.setRollbackOnly();
+                    throw x;
+                }
+            }
+        }
+    }
+
+    /**
+     * Perform the data migration for V0010 (non-multi-tenant schema)
+     */
+    private void doMigrationForV0010() {
```

Comment:
would be nice to have just one doMigrationForV0010 method since the majority of their code is shared

Reply:
Yeah, unfortunately the logic is just different enough. Could be improved, I'm sure, but I'm not going to risk changing anything at this stage.